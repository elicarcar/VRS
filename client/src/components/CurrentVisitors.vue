<template>
  <div>
    <table class="visitor-table" v-show="visitors.length > 0">
      <thead>
        <tr>
          <th></th>
          <th>Visitor</th>
          <th>Email</th>
        </tr>
      </thead>
      <tfoot>
        <tr>
          <th></th>
          <th>Visitor</th>
          <th>Email</th>
        </tr>
      </tfoot>
      <tbody>
        <tr v-for="visitor in visitors" :key="visitor.id">
          <td class="status">
            <span
              :class="[visitor.is_logged ? 'logged-in' : 'logged-out']"
            ></span>
          </td>
          <td>{{ visitor.first_name + visitor.last_name }}</td>
          <td>{{ visitor.email }}</td>
          <td>
            <button class="btn" @click="changeStatus(visitor.id)">
              {{ visitor.is_logged ? 'End' : 'Start' }}
            </button>
          </td>
        </tr>
      </tbody>
    </table>

    <p v-if="!visitors.length">
      There are no visitors at the office right now.
    </p>
  </div>
</template>

<script>
export default {
  name: 'CurrentVisitors',

  props: {
    visitors: Array,
    changeStatus: Function,
  },
}
</script>
