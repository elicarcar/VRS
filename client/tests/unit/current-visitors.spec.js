import { mount, createLocalVue } from '@vue/test-utils'
import { BootstrapVue } from 'bootstrap-vue'
import CurrentVisitors from '@/components/CurrentVisitors.vue'

const localVue = createLocalVue()

localVue.use(BootstrapVue)

describe('CurrentVisitors', () => {
  it('shows visitors if visitors.length > 0', () => {
    const wrapper = mount(CurrentVisitors, {
      propsData: {
        visitors: [],
        changeStatus: jest.fn(),
      },
    })

    const paragraph = wrapper.find('p')
    if (!wrapper.vm.visitors.length) {
      expect(paragraph.text()).toBe(
        'There are no visitors at the office right now.'
      )
    }
  }),
    it("won't show the table if !visitors.length", () => {
      const wrapper = mount(CurrentVisitors, {
        propsData: {
          visitors: [],
          changeStatus: jest.fn(),
        },
      })

      if (!wrapper.vm.visitors.length) {
        expect(wrapper.find('table').isVisible()).toBe(false)
      } else {
        expect(wrapper.find('table').isVisible()).toBe(true)
      }
    })
})
