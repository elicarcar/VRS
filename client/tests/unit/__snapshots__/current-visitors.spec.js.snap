// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentVisitors renders correctly 1`] = `
<div>
  <table class="visitor-table" style="display: none;">
    <thead>
      <tr>
        <th></th>
        <th>Visitor</th>
        <th>Email</th>
        <th></th>
      </tr>
    </thead>
    <tfoot>
      <tr>
        <th></th>
        <th>Visitor</th>
        <th>Email</th>
        <th></th>
      </tr>
    </tfoot>
    <tbody></tbody>
  </table>
  <p data-name="p-no-visitors">
    There are no visitors at the office right now.
  </p>
</div>
`;
